modules/freetype/src/freetype.cpp

```diff
@@ -181,16 +182,39 @@ FreeType2Impl::~FreeType2Impl()
 void FreeType2Impl::loadFontData(String fontFileName, int idx)
 {
     CV_Assert( idx >= 0 );
-    if( mIsFaceAvailable  == true )
+    if ( mIsFaceAvailable  == true )
     {
-        hb_font_destroy (mHb_font);
+        hb_font_destroy(mHb_font);
+        CV_Assert(!FT_Done_Face(mFace));
+    }
+
+    mIsFaceAvailable = false;
+    CV_Assert( !FT_New_Face( mLibrary, fontFileName.c_str(), static_cast<FT_Long>(idx), &mFace ) );
+
+    mHb_font = hb_ft_font_create(mFace, NULL);
+    if ( mHb_font == NULL )
+    {
+        CV_Assert(!FT_Done_Face(mFace));
+        return;
+    }
+    CV_Assert( mHb_font != NULL );
+    mIsFaceAvailable = true;
+}
+
+void FreeType2Impl::loadFontData(uchar* pBuf, size_t bufSize, int idx)
+{
+    CV_Assert( idx >= 0 );
+    if ( mIsFaceAvailable  == true )
+    {
+        hb_font_destroy(mHb_font);
         CV_Assert(!FT_Done_Face(mFace));
     }
 
     mIsFaceAvailable = false;
-    CV_Assert( !FT_New_Face( mLibrary, fontFileName.c_str(), static_cast<FT_Long>(idx), &(mFace) ) );
+    FT_Open_Args args{ FT_OPEN_MEMORY, (FT_Byte*)pBuf, static_cast<FT_Long>(bufSize), nullptr, nullptr, nullptr, 0, nullptr };
+    CV_Assert( !FT_Open_Face(mLibrary, &args, idx, &mFace) );
```

Comment:
Two functions reference different Freetype APIs. It would be better to have one or the other in common.
- Idea1) To use `FT_Open_Face()` for file and memory.
  - both functrions construct `FT_Open_Args` for file and memory.
  - new helper function receives `FT_Open_Args` and `idx`. 
- Idea1) To use `FT_New_Face()` for file, and to use `FT_New_Memory_Face()` for memory.

Reply:
Thank you for your comment. It is OK. I think there is no problem.
In this pull-request, the FreeType function to be called was switched with an additional commit, so I confirmed the intention. thank you very much !!